File: src/snippets/Condition tree/JavaScript/condTree.js
function compare(a, b) {
	let msg;
	print(a, b);

	if(a > b) {
		msg = "a > b";
		a++;
	} else if(a < b) {
		msg = "a < b";
		b++;
	} else {
		msg = "a == b";
		a++;
		b++;
	}
	console.log(msg);
	print(a, b);
	console.log();
}
function print(a, b) {
	console.log(`a = ${a}, b = ${b}`);
}

function main() {
	for(let a = 1; a <= 3; a++)
		for(let b = 1; b <= 3; b++)
			compare(a, b);
}

main()

File: src/snippets/Condition tree/Python/condTree.py
def compare(a: int, b: int):
	printNums(a, b)

	if a > b:
		msg = "a > b"
		a += 1
	elif a < b:
		msg = "a < b"
		b += 1
	else:
		msg = "a == b"
		a += 1
		b += 1

	print(msg)
	printNums(a, b)
	print()

def printNums(a, b):
	print(f"{a = }, {b = }")

def main():
	for a in range(1, 4):
		for b in range(1, 4):
			compare(a, b)

if __name__ == "__main__":
	main()
